### PR TITLE
Add "Switch Transition Direction" to transition's context menu

### LINKED ISSUE #1243: Add "Switch Transition direction" feature

### PR DESCRIPTION
**Pull request by BeckmaR**
Before:
image
Menu:
image
After:
image

Resolves #1243

### CODE REVIEW

File: plugins/org.yakindu.sct.refactoring/src/org/yakindu/sct/refactoring/handlers/AbstractTransitionRefactoringHandler.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package org.yakindu.sct.refactoring.handlers;
```
Comment (andreasmuelder) [inline plugins/org.yakindu.sct.refactoring/src/org/yakindu/sct/refactoring/handlers/AbstractTransitionRefactoringHandler.java]: copyright is missing

File: plugins/org.yakindu.sct.refactoring/src/org/yakindu/sct/refactoring/handlers/AbstractTransitionRefactoringHandler.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import java.util.Collections;
+
+import org.eclipse.jface.viewers.ISelection;
+import org.yakindu.sct.model.sgraph.Transition;
+import org.yakindu.sct.refactoring.refactor.AbstractRefactoring;
+import org.yakindu.sct.ui.editor.editparts.TransitionEditPart;
+
+import com.google.common.collect.Lists;
+
+public abstract class AbstractTransitionRefactoringHandler extends AbstractRefactoringHandler<TransitionEditPart> {
```
Comment (andreasmuelder) [inline plugins/org.yakindu.sct.refactoring/src/org/yakindu/sct/refactoring/handlers/AbstractTransitionRefactoringHandler.java]: author tag is missing

File: plugins/org.yakindu.sct.refactoring/src/org/yakindu/sct/refactoring/handlers/impl/SwitchTransitionDirectionHandler.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package org.yakindu.sct.refactoring.handlers.impl;
```
Comment (andreasmuelder) [inline plugins/org.yakindu.sct.refactoring/src/org/yakindu/sct/refactoring/handlers/impl/SwitchTransitionDirectionHandler.java]: copyright is missing

File: plugins/org.yakindu.sct.refactoring/src/org/yakindu/sct/refactoring/handlers/impl/SwitchTransitionDirectionHandler.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+import org.yakindu.sct.refactoring.handlers.AbstractTransitionRefactoringHandler;
+import org.yakindu.sct.refactoring.refactor.AbstractRefactoring;
+import org.yakindu.sct.refactoring.refactor.impl.SwitchTransitionDirection;
+import org.yakindu.sct.ui.editor.editparts.TransitionEditPart;
+
+public class SwitchTransitionDirectionHandler extends AbstractTransitionRefactoringHandler {
```
Comment (andreasmuelder) [inline plugins/org.yakindu.sct.refactoring/src/org/yakindu/sct/refactoring/handlers/impl/SwitchTransitionDirectionHandler.java]: author tag is missing

File: plugins/org.yakindu.sct.refactoring/src/org/yakindu/sct/refactoring/refactor/impl/SwitchTransitionDirection.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
+package org.yakindu.sct.refactoring.refactor.impl;
```
Comment (andreasmuelder) [inline plugins/org.yakindu.sct.refactoring/src/org/yakindu/sct/refactoring/refactor/impl/SwitchTransitionDirection.java]: copyright is missing

File: plugins/org.yakindu.sct.refactoring/src/org/yakindu/sct/refactoring/refactor/impl/SwitchTransitionDirection.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+import org.yakindu.sct.model.sgraph.Transition;
+import org.yakindu.sct.model.sgraph.Vertex;
+import org.yakindu.sct.refactoring.refactor.AbstractRefactoring;
+import org.yakindu.sct.ui.editor.editparts.TransitionEditPart;
+
+/* 
+ * TODO: This is not really a Refactoring. A Refactoring changes stuff without changing what it does, which does
+ * not apply on this "refactoring". There does not seem to be a better place for this at the moment, so when something like
+ * Modifications exists, this should be moved there.
+ */
```
Comment (andreasmuelder) [inline plugins/org.yakindu.sct.refactoring/src/org/yakindu/sct/refactoring/refactor/impl/SwitchTransitionDirection.java]: You are right this is not a refactoring but a model modification. In fact, we expose an API for model modifications that is used for example for the 'add outgoing state' action.
Take a look at org.yakindu.sct.ui.editor.modifications.ISemanticModification and AbstractSemanticModification.
There is also an adapter for both APIs RefactoringBasedModification that is used for Inline and Extract Subdiagram.

**Comment by BeckmaR** [comment]
Closing, changes merged into issue_1425 (see #1425).

**Pull request closed without merging**